lib/barbeque/runner.rb

```diff
@@ -0,0 +1,51 @@
+require 'barbeque/config'
+require 'barbeque/exception_handler'
+require 'barbeque/execution_log'
+require 'barbeque/message_handler'
+require 'barbeque/message_queue'
+
+module Barbeque
+  class Runner
```

Comment:
In that case, what methods are needed for Executor would be helpful information for reading barbeque or writing another Executor backend.